server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -191,21 +199,35 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
      * @return List of Entitlements
      * @throws AutobindDisabledForOwnerException when an autobind attempt is made and the owner
      *         has it disabled.
+     * @throws AutobindHypervisorDisabledException when an autobind attempt is made on a hypervisor
+     *         and the owner has it disabled.
      */
     @Transactional
     public List<Entitlement> bindByProducts(AutobindData data, boolean force)
-        throws AutobindDisabledForOwnerException {
+        throws AutobindDisabledForOwnerException, AutobindHypervisorDisabledException {
         Consumer consumer = data.getConsumer();
         Owner owner = data.getOwner();
-
-        if ((!consumer.isDev() && owner.isAutobindDisabled()) || owner.isContentAccessEnabled()) {
+        ConsumerType type = this.consumerTypeCurator.getConsumerType(consumer);
+        if (!consumer.isDev() && (owner.isAutobindDisabled() ||
+            owner.isContentAccessEnabled() ||
+            (type != null && type.isType(ConsumerType.ConsumerTypeEnum.HYPERVISOR) &&
+            owner.isAutobindHypervisorDisabled()))) {
             String caMessage = owner.isContentAccessEnabled() ?
                 " because of the content access mode setting" : "";
-            log.info("Skipping auto-attach for consumer '{}'. Auto-attach is disabled for owner {}{}",
-                consumer.getUuid(), owner.getKey(), caMessage);
-            throw new AutobindDisabledForOwnerException(i18n.tr(
-                "Auto-attach is disabled for owner \"{0}\"{1}.",
-                owner.getKey(), caMessage));
+            String hypMessage = owner.isAutobindHypervisorDisabled() ?
+                " because of the hypervisor autobind setting" : "";
+            log.info("Skipping auto-attach for consumer '{}'. Auto-attach is disabled for owner {}{}{}",
+                consumer.getUuid(), owner.getKey(), caMessage, hypMessage);
+            if (!"".equals(hypMessage)) {
+                throw new AutobindHypervisorDisabledException(i18n.tr(
+                    "Auto-attach is disabled for owner \"{0}\"{1}.",
+                    owner.getKey(), hypMessage));
+            }
+            else {
+                throw new AutobindDisabledForOwnerException(i18n.tr(
```

Comment:
We can save a few lines of code here by using a tertiary op:

```throw new AutobindHypervisorDisabledException(..., !hypMessage.isEmpty() ? hypMessage : caMessage);```

Reply:
Those are different exceptions by design. Not sure we can use this. I'll change that condition.

Reply:
Ahh, my brain autocorrected the exception names. Disregard.